src/app/model/ChessEngine.ts

```diff
@@ -37,8 +38,9 @@ export class ChessEngine implements ChessModel {
     };
     const pieceWasCaptured = this.pieceWasCaptured(squareTo, chosenPiece);
 
+    const pawnPromotionWasEnabled = chosenPiece instanceof Pawn ? this.pawnPromotionWasEnabled(chosenPiece, squareTo) : undefined;
     this.onPieceWasMoved(pieceWasMoved);
-    return pieceWasCaptured ? [pieceWasMoved, pieceWasCaptured] : [pieceWasMoved];
+    return [pieceWasMoved, pieceWasCaptured, pawnPromotionWasEnabled].filter(isDefined);
```

Comment:
Spoko roziwazanie z tym filtrowaniem :) Moznaby ewentualnie nadac nazwe jakas bardziej eventowa np. 'hasOccurred', czyli, ze wydarzyly sie.

Reply:
To bym musiała to ubrać w jeszcze osobną metodę, bo isDefined jest używane jeszcze w innym miejscu i trochę chyba nie będzie pasowało hasOccurred.

Reply:
done